mypyc/genops.py

```diff
@@ -701,12 +702,101 @@ def visit_assignment_stmt(self, stmt: AssignmentStmt) -> Value:
         self.assign(lvalue, stmt.rvalue)
         return INVALID_VALUE
 
+    def translate_in_place_op(self, op: str) -> Optional[str]:
+        op_map = {
+            '+': '__iadd__',
+            '-': '__isub__',
+            '*': '__imul__',
+            '@': '__imatmul__',
+            '/': '__itruediv__',
+            '//': '__ifloordiv__',
+            '%': '__imod__',
+            '**': '__ipow__',
+            '<<': '__ilshift__',
+            '>>': '__irshift__',
+            '&': '__iand__',
+            '^': '__ixor__',
+            '|': '__ior__'
+        }
+        return op_map.get(op)
```

Comment:
If we want to do this optimization of ops on RInstance values, I think that the best place to do it would be in `binary_op`. We'd want to cover the non-in-place version as well.

We don't properly support implementing these methods yet, though. (We don't fill out the structures required for cpython to properly call them.)